server/storage/schema/schema_test.go

```diff
@@ -182,15 +182,14 @@ func TestMigrate(t *testing.T) {
 			expectVersion: nil,
 		},
 		{
-			name:          "Downgrading v3.6 to v3.5 fails if there are newer WAL entries",
+			name:          "Downgrading v3.6 to v3.5 works even if there is ClusterVersionSetRequest WAL entries with 3.6 clusterVersion",
 			version:       version.V3_6,
 			targetVersion: version.V3_5,
 			walEntries: []etcdserverpb.InternalRaftRequest{
 				{ClusterVersionSet: &membershippb.ClusterVersionSetRequest{Ver: "3.6.0"}},
 			},
-			expectVersion:  &version.V3_6,
-			expectError:    true,
-			expectErrorMsg: "cannot downgrade storage, WAL contains newer entries",
+			expectVersion: nil, // 3.5 doesn't have field `storageVersion`, so it should be nil.
+			expectError:   false,
 		},
```

Comment:
I think we should have a test case that confirms that downgrade cannot happen if there is newer WAL entry. This might be tricky for v3.6 -> v3.5 as we don't have such entry. My original idea was to add a fake entry to wal proto definition that can be used for testing.

Reply:
added a TODO, which will be resolved in a followup PR. thx

Reply:
SG